livre/lib/pipeline/RenderPipeline.cpp

```diff
@@ -82,40 +86,104 @@ struct RenderPipeline::Impl
         }
     }
 
-    void createSyncPipeline( PipeFilter& renderFilter,
-                             PipeFilter& histogramFilter,
-                             Pipeline& renderPipeline,
-                             Pipeline& uploadPipeline ) const
+    void createSyncPipeline( const RenderParams& renderParams,
+                             PipeFilter& sendHistogramFilter,
+                             Renderer& renderer,
+                             NodeAvailability& availability ) const
     {
-        PipeFilter visibleSetGenerator =
-                renderPipeline.add< VisibleSetGeneratorFilter >(
-                    "VisibleSetGenerator", _dataSource );
+        PipeFilterT< VisibleSetGeneratorFilter > visibleSetGenerator( "VisibleSetGenerator", _dataSource );
 
-        createAndConnectUploaders( uploadPipeline,
-                                   visibleSetGenerator,
-                                   renderFilter );
+        visibleSetGenerator.getPromise( "Frustum" ).set( renderParams.frameInfo.frustum );
+        visibleSetGenerator.getPromise( "Frame" ).set( renderParams.frameInfo.timeStep );
+        visibleSetGenerator.getPromise( "DataRange" ).set( renderParams.renderDataRange );
+        visibleSetGenerator.getPromise( "Params" ).set( renderParams.vrParams );
+        visibleSetGenerator.getPromise( "Viewport" ).set( renderParams.pixelViewPort );
+        visibleSetGenerator.getPromise( "ClipPlanes" ).set( renderParams.clipPlanes );
 
-        for( size_t i = 0; i < nUploadThreads; ++i )
+        visibleSetGenerator.execute();
+
+        sendHistogramFilter.getPromise( "RelativeViewport" ).set( renderParams.viewport );
+        sendHistogramFilter.getPromise( "Id" ).set( renderParams.frameInfo.frameId );
+
+        const livre::UniqueFutureMap portFutures( visibleSetGenerator.getPostconditions( ));
+        const auto& nodeIds = renderer.order( portFutures.get< NodeIds >( "VisibleNodes" ),
+                                              renderParams.frameInfo.frustum );
+
+        const uint32_t maxNodesPerPass =
+                renderParams.vrParams.getMaxGPUCacheMemoryMB( ) * ( 1024.f * 1024.f ) /
```

Comment:
use LB_1MB  instead of 1024.f * 1024.f and renderParams.vrParams.getMaxGPUCacheMemoryMB()